recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-core/images/core-image-thin-initramfs.bbappend

```diff
@@ -29,17 +31,57 @@ python __anonymous () {
         d.appendVar("IMAGE_INSTALL", " domu domu-run domu-install-artifacts")
 }
 
-generate_uboot_image() {
-    ${STAGING_BINDIR_NATIVE}/uboot-mkimage -A arm64 -O linux -T ramdisk -C gzip -n "uInitramfs" \
-        -d ${DEPLOYDIR}-image-complete/${IMAGE_LINK_NAME}.cpio.gz ${DEPLOY_DIR_IMAGE}/${IMAGE_NAME}${IMAGE_NAME_SUFFIX}.cpio.gz.uInitramfs
-    ln -sfr  ${DEPLOY_DIR_IMAGE}/${IMAGE_NAME}${IMAGE_NAME_SUFFIX}.cpio.gz.uInitramfs ${DEPLOY_DIR_IMAGE}/uInitramfs
+IMAGE_ROOTFS_SIZE = "65535"
+
+WKS_FILE = "prod-devel.wks"
+IMAGE_FSTYPES += "wic.xz wic.bmap"
+
+#Do not update /etc/fstab
+WIC_CREATE_EXTRA_ARGS_append = " --no-fstab-update"
+
+#Folder where we collect boot partition contents
+DOM0_BOOT_FILES_DIR = "${DEPLOY_DIR_IMAGE}/bootfiles"
+
+#Tell wic to take boot files from the file system
+WIC_CREATE_EXTRA_ARGS_append = " --rootfs-dir rootfs_boot=${DOM0_BOOT_FILES_DIR}"
+
+addtask collect_artifacts after do_image_cpio before do_image_wic
+do_collect_artifacts() {
+    # Generate u-boot image for initramfs
+     ${STAGING_BINDIR_NATIVE}/uboot-mkimage -A arm64 -O linux -T ramdisk -C gzip -n "uInitramfs" \
+         -d ${DEPLOYDIR}-image-complete/${IMAGE_LINK_NAME}.cpio.gz ${DEPLOY_DIR_IMAGE}/${IMAGE_NAME}${IMAGE_NAME_SUFFIX}.cpio.gz.uInitramfs
+     ln -sfr  ${DEPLOY_DIR_IMAGE}/${IMAGE_NAME}${IMAGE_NAME_SUFFIX}.cpio.gz.uInitramfs ${DEPLOY_DIR_IMAGE}/uInitramfs
+    # Collect all the boot files
+    rm -rf ${DOM0_BOOT_FILES_DIR}
+    install -d ${DOM0_BOOT_FILES_DIR}
+
+    # Install bootloader files built by DomD into Dom0 deploy dir,
+    # so all the artifacts required for flashing are at the same place
+
+    find ${XT_DIR_ABS_SHARED_BOOT_DOMD}/* -type f | xargs -I{} cp -f {} ${DEPLOY_DIR_IMAGE}
+
+    # Copy all the boot files generated by DomD...
+    if [ -f ${XT_DIR_ABS_SHARED_BOOT_DOMD}/dom0.dtb ]; then
+        cp -fL ${XT_DIR_ABS_SHARED_BOOT_DOMD}/dom0*.dtb ${DOM0_BOOT_FILES_DIR}
```

Comment:
why do you need to copy all by mask? I think dom0.dtb is enough
IMO, we just need to follow the symlink while copying

Reply:
I am not sure if name will be permanent , but  if yes - i do

Reply:
fixed